api/envoy/annotations/deprecation.proto

```diff
@@ -0,0 +1,15 @@
+syntax = "proto3";
+
+package envoy.annotations;
+
+import "google/protobuf/descriptor.proto";
+
+// Magic number in this file derived from top 28bit of SHA256 digest of "envoy.annotation.do_not_use"
+extend google.protobuf.FieldOptions {
```

Comment:
Do you mind adding some comments about what these annotations do? I'm a little confused what the proposed flow is?